pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,181 @@
+	GetDeployment(ctx kapi.Context, name string) (*kapi.ReplicationController, error)
+	WatchDeployments(ctx kapi.Context, label labels.Selector, field fields.Selector, rv string) (watch.Interface, error)
+}
+
+// Client provides an implementation of DeploymentClient
+type Client struct {
+	GetDcFn   func(ctx kapi.Context, name string) (*deployapi.DeploymentConfig, error)
+	GetRcFn   func(ctx kapi.Context, name string) (*kapi.ReplicationController, error)
+	WatchRcFn func(ctx kapi.Context, label labels.Selector, field fields.Selector, rv string) (watch.Interface, error)
+}
+
+// GetDeploymentConfig returns the deploymentConfig with the provided context and name
+func (c Client) GetDeploymentConfig(ctx kapi.Context, name string) (*deployapi.DeploymentConfig, error) {
+	return c.GetDcFn(ctx, name)
+}
+
+// GetDeployment returns the deployment with the provided context and name
+func (c Client) GetDeployment(ctx kapi.Context, name string) (*kapi.ReplicationController, error) {
+	return c.GetRcFn(ctx, name)
+}
+
+// WatchDeployments watches the deployments with the provided context, label/field selectors, and resource version
+func (c Client) WatchDeployments(ctx kapi.Context, label labels.Selector, field fields.Selector, rv string) (watch.Interface, error) {
+	return c.WatchRcFn(ctx, label, field, rv)
+}
+
+type podGetter struct {
+	podsNamespacer kclient.PodsNamespacer
+}
+
+// Get is responsible for retrieving the deployer pod
+func (g *podGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return g.podsNamespacer.Pods(ns).Get(name)
+}
+
+const defaultTimeout time.Duration = 10 * time.Second
+
+// NewREST creates a new REST for DeployLogs. It accepts a DeploymentClient and a pod
+// client to get necessary attributes to assemble the URL to which the request shall be
+// redirected in order to get the deployment logs.
+func NewREST(d DeploymentClient, pn kclient.PodsNamespacer, connectionInfo kclient.ConnectionInfoGetter) *REST {
+	return &REST{
+		Deploy:         d,
+		PodGetter:      &podGetter{pn},
+		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
+	}
+}
+
+// REST implements GetterWithOptions
+var _ = rest.GetterWithOptions(&REST{})
+
+// Get returns a streamer resource with the contents of the deployment log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	deployLogOpts, ok := opts.(*deployapi.DeploymentLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("did not get an expected options.")
+	}
+
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
+	}
+
+	deployment, err := r.Deploy.GetDeployment(ctx, deployutil.LatestDeploymentNameForConfig(config))
+	if err != nil {
+		// TODO: Fallback to latest-1?
```

Comment:
Based on a comment in IRC, I was totally wrong about this... I think what you originally had was right. If you fall back at all, then you might get logs for N-1 if N completed since deployer pods are deleted upon successful completion.